src/Extension/Events/CodeRunEvaluationProduced.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.DotNet.Interactive.Commands;
+
+namespace Extension.Events
+{
+    public class CodeRunEvaluationProduced : EvaluationProduced
```

Comment:
What is the goal of using inheritance here?

Reply:
Not very necessary at the moment. It does reduce duplicated Evaluation field, which is shared among all EvaluationProduced objects.

This might be useful when manipulating multiple types of evaluation produced. Although we could also accomplish the same thing by folding this information as a property of the Evaluation class.

I'll keep this inheritance out of this pr